persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -68,100 +68,130 @@ class MessageDAOImpl(private val queries: MessagesQueries, private val conversat
      */
     @Suppress("ComplexMethod", "LongMethod")
     private fun insertInDB(message: MessageEntity) {
-        queries.insertMessage(
-            id = message.id,
-            conversation_id = message.conversationId,
-            date = message.date,
-            sender_user_id = message.senderUserId,
-            sender_client_id = if (message is MessageEntity.Regular) message.senderClientId else null,
-            visibility = message.visibility,
-            status = message.status,
-            content_type = contentTypeOf(message.content)
-        )
-        when (val content = message.content) {
-            is MessageEntityContent.Text -> {
-                queries.insertMessageTextContent(
-                    message_id = message.id,
-                    conversation_id = message.conversationId,
-                    text_body = content.messageBody
-                )
-                content.mentions.forEach {
-                    queries.insertMessageMention(
+        if (!updateIdIfAlreadyExists(message)) {
```

Comment:
Just to be a bit more efficient, we need to check if it already exists only if the sender is self

Reply:
Sure, added it to the code 👍